### PR TITLE
feat(artifacts): device.takeScreenshot() returns a temporary screenshot path

### PR DESCRIPTION
**Pull request by noomorph**
- [x] This change has been discussed in issue #1712 and #1362 and the solution has been agreed upon with maintainers.

---

**Description:**

**TODOs:**

- [x] fix unit tests
- [x] restore coverage
- [x] merge in documentation `--take-screenshots none` and `--take-screenshots manual` modes (`none` will behave like `manual` nowadays).
- [x] make a round of regression testing on Jest
- [x] make a round of regression testing on Mocha

### CODE REVIEW

**Comment by rotemmiz** [comment]
Why do we need to choose `--take-screenshots none` or `--take-screenshots manual`?
What blocks us from supporting both of the following?
1. `--take-screenshots all/none/failing` as we currently have, this is implicit for every test case.
2. `device.takeScreenshot` is explicit, and is totally disconnected from anything the user sets in cli params.

**Comment by LeoNatan** [comment]
@rotemmiz The problem was that in the default mode, `none`, manual screenshots were not being saved. 

I think manual screenshots should be tied up to the `take-screenshots` setting. That should only be for test life-cycle, while user screenshots should always be included.

> probably they will be merged, and `none` will behave like `manual` nowadays

This sounds like a good idea.

**Comment by LeoNatan** [comment]
Might be a good idea to note in docs that the returned path is temporary and will be invalid once the test is complete.

**Comment by noomorph** [comment]
@LeoNatan , this is already done in docs in this pull request.

File: docs/APIRef.Artifacts.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ Artifacts are disabled by default. Two things are required to enable them:
 * To record `.mp4` test run videos, add `--record-videos all` (or `--record-videos failing`, if you want to keep video recordings only for failing tests).
 * To record `.dtxrec` (Detox Instruments recordings) for each test, add `--record-performance all`. To open those recordings, you'll need [Detox Instruments](https://github.com/wix/DetoxInstruments). **NOTE:** only iOS is supported.
 * To take `.png` screenshots before and after each test, add `--take-screenshots all` (or `--take-screenshots failing`, if you want to keep only screenshots of failing tests).  
-Alternatively, you might leverage `await device.takeScreenshot('your screenshot name')` API to have manual control over taking the screenshots.  
-To disable screenshots subsystem forcibly, use `--take-screenshots none`.
+Alternatively, you might leverage [device.takeScreenshot()](APIRef.DeviceObjectAPI.md#devicetakescreenshotname) API for the manual control.
```
Comment (LeoNatan) [inline docs/APIRef.Artifacts.md]: "you might leverage the …"
"API for ~~the~~ manual control"
🙂

File: docs/APIRef.DeviceObjectAPI.md [hunk 2]
```diff
@@ -325,24 +325,34 @@ if (device.getPlatform() === 'ios') {
 ```
 
 ### `device.takeScreenshot(name)`
+
 Takes a screenshot on the device and schedules putting it to
 the [artifacts folder](APIRef.Artifacts.md#enabling-artifacts) upon
-completion of the current test. Consider the example below:
+completion of the current test.
+
+Returns the temporary path to the screenshot.
```
Comment (LeoNatan) [inline docs/APIRef.DeviceObjectAPI.md]: ~~the~~ **a** temporary path

**Comment by LeoNatan** [comment]
@noomorph Sorry, missed it.

**Comment by noomorph** [comment]
@Gongreg, the pull request has landed. We'll release a version tomorrow, I think.

**Comment by Gongreg** [comment]
@noomorph, awesome! :)

**Pull request merged**